pkg/cmd/pulumi/state_move_test.go

```diff
@@ -630,3 +630,64 @@ func TestMoveUnknownResource(t *testing.T) {
 	assert.Contains(t, stdout.String(), "warning: Resource not-a-urn not found in source stack")
 	assert.Equal(t, 3, len(sourceSnapshot.Resources)) // No resources should be moved
 }
+
+func TestMoveProvider(t *testing.T) {
+	t.Parallel()
+
+	providerURN := resource.NewURN("sourceStack", "test", "", "pulumi:providers:a", "default_1_0_0")
+	sourceResources := []*resource.State{
+		{
+			URN:    providerURN,
+			Type:   "pulumi:providers:a::default_1_0_0",
+			ID:     "provider_id",
+			Custom: true,
+		},
+		{
+			URN:      resource.NewURN("sourceStack", "test", "d:e:f", "a:b:c", "name"),
+			Type:     "a:b:c",
+			Provider: string(providerURN) + "::provider_id",
+		},
+		{
+			URN:      resource.NewURN("sourceStack", "test", "d:e:f", "a:b:c", "name2"),
+			Type:     "a:b:c",
+			Provider: string(providerURN) + "::provider_id",
+			Parent:   resource.NewURN("sourceStack", "test", "d:e:f", "a:b:c", "name"),
+		},
+	}
+
+	ctx := context.Background()
+	tmpDir := t.TempDir()
+	t.Cleanup(func() {
+		os.RemoveAll(tmpDir)
+	})
```

Comment:
Isn't it automatically cleaned up? https://pkg.go.dev/testing#T.TempDir

Reply:
Oh yes, it should be!